File: 22 08 25/integer2roman.cs
using System;

public class Solution {
    public string IntToRoman(int num) {
        int[] val = {1000, 900, 500, 400,
                     100, 90, 50, 40,
                     10, 9, 5, 4,
                     1};
        string[] syms = {"M", "CM", "D", "CD",
                         "C", "XC", "L", "XL",
                         "X", "IX", "V", "IV",
                         "I"};

        System.Text.StringBuilder roman = new System.Text.StringBuilder();

        for (int i = 0; i < val.Length; i++) {
            while (num >= val[i]) {
                num -= val[i];
                roman.Append(syms[i]);
            }
        }

        return roman.ToString();
    }
}

File: 22 08 25/integer2roman.c
#include <stdio.h>
#include <string.h>

char* intToRoman(int num, char* result) {
    int val[] = {1000, 900, 500, 400,
                 100, 90, 50, 40,
                 10, 9, 5, 4,
                 1};
    char* syms[] = {"M", "CM", "D", "CD",
                    "C", "XC", "L", "XL",
                    "X", "IX", "V", "IV",
                    "I"};

    result[0] = '\0';  // initialize result string

    for (int i = 0; i < 13; i++) {
        while (num >= val[i]) {
            num -= val[i];
            strcat(result, syms[i]);
        }
    }

    return result;
}